CustomErrors/CustomErrors_test.go

```diff
@@ -65,17 +65,95 @@ func TestErrorImplements_errorInterface(t *testing.T) {
 	var _ error = &Error{}
 }
 
-func TestErrorErrorMethod(t *testing.T) {
-	expected := `CustomErrors.Error{Type:AccessViolation, Message:Testing}`
-	var err error = Error{Type: AccessViolation, Message: "Testing"}
-	if err.Error() != expected {
-		t.Errorf("err.Error() returned %v while expected %v", err.Error(), expected)
+func TestErrorTypeStringer(t *testing.T) {
+
+	testCases := []struct {
+		TestAlias      string
+		ErrorType      ErrorType
+		ExpectedString string
+	}{
+		{
+			TestAlias:      "BasicError",
+			ErrorType:      BasicError,
+			ExpectedString: "BasicError",
+		},
+		{
+			TestAlias:      "InvalidOperation",
+			ErrorType:      InvalidOperation,
+			ExpectedString: "InvalidOperation",
+		},
+		{
+			TestAlias:      "InvalidArgument",
+			ErrorType:      InvalidArgument,
+			ExpectedString: "InvalidArgument",
+		},
+		{
+			TestAlias:      "AccessViolation",
+			ErrorType:      AccessViolation,
+			ExpectedString: "AccessViolation",
+		},
+		{
+			TestAlias:      "Nonsupported",
+			ErrorType:      Nonsupported,
+			ExpectedString: "Nonsupported",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testAlias := testCase.TestAlias
+		errorType := testCase.ErrorType
+		expectedString := testCase.ExpectedString
+
+		testFn := func(t *testing.T) {
+
+			actualString := fmt.Sprintf("%s", errorType)
```

Comment:
According to [this](https://golang.org/pkg/fmt/) `%s	the uninterpreted bytes of the string or slice` is exactly what I want to check. In other words I want s of type ErrorType to be treated by *print(f) as a string